File: 1-TwoSum/1-TwoSum.java
// Last updated: 04/04/2026, 23:39:16
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++) {
            int key = target - nums[i];
            if(map.containsKey(key)) {
                return new int[]{i, map.get(key)};
            }
            map.put(nums[i], i);
        } 
        return new int[]{0, 0};
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 04/04/2026, 23:39:19
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hm;
        for(int i=0;i<nums.size();i++){
            hm.insert({nums[i],i});
        }
        for(int i=0;i<nums.size();i++){
            int find=target-nums[i];
            if(hm.count(find)==true && hm[find]!=i){
                return {i,hm[find]};
            }
        }
        return {};
    }
};